packages/golden-layout/src/controls/DragSource.ts

```diff
@@ -0,0 +1,71 @@
+import $ from 'jquery';
+import type { ItemConfigType } from '../config';
+import type LayoutManager from '../LayoutManager';
+import { DragListener } from '../utils';
+import DragProxy from './DragProxy';
+
+/**
+ * Allows for any DOM item to create a component on drag
+ * start tobe dragged into the Layout
+ *
+ * @param element
+ * @param itemConfig the configuration for the contentItem that will be created
+ * @param layoutManager
+ */
+export default class DragSource {
+  _element: JQuery<HTMLElement>;
+  _itemConfig: ItemConfigType | (() => ItemConfigType);
+  _layoutManager: LayoutManager;
+  _dragListener: DragListener;
+
+  constructor(
+    element: JQuery<HTMLElement>,
+    itemConfig: ItemConfigType | (() => ItemConfigType),
+    layoutManager: LayoutManager
+  ) {
+    this._element = element;
+    this._itemConfig = itemConfig;
+    this._layoutManager = layoutManager;
+
+    this._dragListener = this._createDragListener(); // This assignment makes TS happy that this is assigned in the constructor and not undefined
```

Comment:
It was because the original code was

```
this._layoutManager = layoutManager;
_createDragListener();
```

And the `_createDragListener` method assigned `this._dragListener`. So I can remove the comment because it's not really an out of place assignment. I think initially I might have told TS that it exists and to not complain but then fixed it.